### PR TITLE
Make group slug nullable and enable unique_index

### PR DESCRIPTION
**Pull request by bduffany**

### CODE REVIEW

**Comment by bduffany** [comment]
Added a migration to ensure all empty strings are first converted to `NULL` so that the unique constraint is valid. Here's the code I used to test the migration: https://gist.github.com/bduffany/19477971e7ab585c60aa2349d32b9511

File: server/buildbuddy_server/buildbuddy_server.go [hunk 1]
```diff
@@ -124,8 +128,10 @@ func (s *BuildBuddyServer) GetGroup(ctx context.Context, req *grpb.GetGroupReque
...
 		return nil, status.UnimplementedError("Not Implemented")
 	}
-	group := &tables.Group{
-		URLIdentifier: req.GetUrlIdentifier(),
+	group := &tables.Group{}
+	urlIdentifier := strings.TrimSpace(req.GetUrlIdentifier())
+	if urlIdentifier != "" {
+		group.URLIdentifier = &urlIdentifier
 	}
 	if err := userDB.FillGroup(ctx, group); err != nil {
```
Comment (tylerwilliams) [inline server/buildbuddy_server/buildbuddy_server.go:136]: Took me a minute to grok this -- I might be a dummy but FillGroup is -- taking input and output in the same argument which is error prone.

Let's remove FillGroup and replace it with GetGroupByID and/or GetGroupByURLIdentifier?

**Review by tylerwilliams — CHANGES_REQUESTED** [review]

File: server/buildbuddy_server/buildbuddy_server.go (hunk 1 above)
Comment (bduffany) [reply server/buildbuddy_server/buildbuddy_server.go:136]: Sure, will do that in a separate PR since it's unrelated. Don't want to block this PR on it

Comment (tylerwilliams) [reply server/buildbuddy_server/buildbuddy_server.go:136]: SGTM.

**Pull request merged**